lib/fluent/supervisor.rb

```diff
@@ -1196,6 +1208,35 @@ def build_system_config(conf)
       system_config
     end
 
+    def build_additional_configuration
+      conf = nil
+      if @system_config.config_include_dir&.empty?
+        $log.info :supervisor, 'default configuration include directory was disabled'
+        return nil
+      end
+      begin
+        Find.find(@system_config.config_include_dir) do |path|
+          next if File.directory?(path)
+          next unless [".conf", ".yaml", ".yml"].include?(File.extname(path))
+          # NOTE: both types of normal config (.conf) and YAML will be loaded.
+          # Thus, it does not care whether @config_path is .conf or .yml.
+          $log.info :supervisor, 'loading additional configuration file', path: path
+          element = Fluent::Config.build(config_path: path,
+                                         encoding: @conf_encoding,
+                                         use_v1_config: @use_v1_config,
+                                         type: :guess)
+          if conf
+            conf += element
+          else
+            conf = element
+          end
```

Comment:
How about using `yield` to avoid managing `conf`?

```rb
yield Fluent::Config.build(config_path: path,
                           encoding: @conf_encoding,
                           use_v1_config: @use_v1_config,
                           type: :guess) 
```

```rb
build_additional_configurations do |additional_conf|
  @conf += additional_conf
end
```

* Codes become simpler.
* if `return` without `yield`, then the block is not called, so we can omit `nil` check too.
* build_additional_configuration**s** would be better?

Reply:
Thanks, good idea.